### PR TITLE
Remove zlib from requirements script and instruction files

### LINKED ISSUE #105373: Remove system zlib packages from unix native dependencies
Follow up of https://github.com/dotnet/runtime/pull/105371

We can't yet remove the system zlib packages from the package installation branches in `install-native-dependencies.sh` because we don't yet consume an SDK version that includes zlib-ng.

If we remove them from this script, and we attempt to run it in a new machine, then try to build runtime with:

```
eng/install-native-dependencies.sh
./build.sh clr+libs
```

We see this error:
```
cdac-build-tool -> /home/carlos/runtime/artifacts/bin/coreclr/linux.x64.Debug/cdac-build-tool/cdac-build-tool.dll
  /usr/bin/ld.bfd: cannot find -lz: No such file or directory
clang : error : linker command failed with exit code 1 (use -v to see invocation) [/home/carlos/runtime/src/native/managed/cdacreader/src/cdacreader.csproj]
/home/carlos/.nuget/packages/microsoft.dotnet.ilcompiler/9.0.0-preview.7.24371.6/build/Microsoft.NETCore.Native.targets(368,5): error MSB3073: The command ""clang" "/home/carlos/runtime/artifacts/obj/cdacreader/Debug/net9.0/linux-x64/native/libcdacreader.o" -o "/home/carlos/runtime/artifacts/bin/cdacreader/Debug/net9.0/linux-x64/native/libcdacreader.so" -Wl,--version-script=/home/carlos/runtime/artifacts/obj/cdacreader/Debug/net9.0/linux-x64/native/libcdacreader.exports -Wl,--export-dynamic -Wl,-soname=cdacreader.so -gz=zlib -fuse-ld=bfd /home/carlos/.nuget/packages/runtime.linux-x64.microsoft.dotnet.ilcompiler/9.0.0-preview.7.24371.6/sdk/libbootstrapperdll.o /home/carlos/.nuget/packages/runtime.linux-x64.microsoft.dotnet.ilcompiler/9.0.0-preview.7.24371.6/sdk/libRuntime.WorkstationGC.a /home/carlos/.nuget/packages/runtime.linux-x64.microsoft.dotnet.ilcompiler/9.0.0-preview.7.24371.6/sdk/libeventpipe-disabled.a /home/carlos/.nuget/packages/runtime.linux-x64.microsoft.dotnet.ilcompiler/9.0.0-preview.7.24371.6/sdk/libRuntime.VxsortEnabled.a /home/carlos/.nuget/packages/runtime.linux-x64.microsoft.dotnet.ilcompiler/9.0.0-preview.7.24371.6/sdk/libstandalonegc-disabled.a /home/carlos/.nuget/packages/runtime.linux-x64.microsoft.dotnet.ilcompiler/9.0.0-preview.7.24371.6/sdk/libstdc++compat.a /home/carlos/.nuget/packages/runtime.linux-x64.microsoft.dotnet.ilcompiler/9.0.0-preview.7.24371.6/framework/libSystem.Native.a /home/carlos/.nuget/packages/runtime.linux-x64.microsoft.dotnet.ilcompiler/9.0.0-preview.7.24371.6/framework/libSystem.IO.Compression.Native.a /home/carlos/.nuget/packages/runtime.linux-x64.microsoft.dotnet.ilcompiler/9.0.0-preview.7.24371.6/framework/libSystem.Net.Security.Native.a /home/carlos/.nuget/packages/runtime.linux-x64.microsoft.dotnet.ilcompiler/9.0.0-preview.7.24371.6/framework/libSystem.Security.Cryptography.Native.OpenSsl.a -g -Wl,-rpath,'$ORIGIN' -Wl,--build-id=sha1 -Wl,--as-needed -Wl,-e0x0 -pthread -ldl -lz -lrt -lm -shared -Wl,-z,relro -Wl,-z,now -Wl,--eh-frame-hdr -Wl,--discard-all -Wl,--gc-sections" exited with code 1. [/home/carlos/runtime/src/native/managed/cdacreader/src/cdacreader.csproj]
```

Once we start consuming an SDK version that includes zlib-ng, we can remove those 3 references from the script: `zlib1g-dev`, `zlib-devel` and `zlib-dev`.

https://github.com/dotnet/runtime/blob/c5b2dcd2967390a2ccc4546cdf3dbf5745d0b0d2/eng/install-native-dependencies.sh#L26-L33

And also remove mentions of zlib in the instructions:

https://github.com/dotnet/runtime/blob/main/docs/workflow/requirements/linux-requirements.md

### PR DESCRIPTION
**Pull request by carlossanlop**
Fixes https://github.com/dotnet/runtime/issues/105373

Follow up of: https://github.com/dotnet/runtime/pull/105371

**Important**: We should merge this when runtime consumes a version of the SDK that already includes the zlib-ng changes, otherwise the script will break.

### CODE REVIEW

**Review by jkotas — APPROVED** [review]

**Review by stephentoub — APPROVED** [review]

**Comment by jkotas** [comment]
This was marked NO MERGE. It can be merged only after we switch to P7 SDK. Merging it now is going to break people and systems that are using install-native-dependencies.sh as part of their workflow.

**Comment by stephentoub** [comment]
Sorry, I missed the label.

**Comment by stephentoub** [comment]
@ViktorHofer, is there a way to make not having a NO-MERGE label a criteria of mergability on GH, just like we have a requirement for an approver and the build analysis leg to have passed?

**Comment by jkoritzinsky** [comment]
I've opened an issue to track adding that.

**Pull request merged**
